lhotse/dataset/collation.py

```diff
@@ -315,6 +315,44 @@ def collate_multi_channel_audio(cuts: CutSet) -> torch.Tensor:
     return audio
 
 
+def remove_pad_tracks(cuts):
+    for cut in cuts:
+        tracks = cut.tracks
+        tracks_nopad = [t for t in tracks if not isinstance(t.cut, PaddingCut)]
+        cut.tracks = tracks_nopad
+    return cuts
+
+
+def collate_multi_channel_audio_rmpad(cuts: CutSet) -> torch.Tensor:
+    """
+    Load audio samples for all the cuts and return them as a batch in a torch tensor.
+    The cuts have to be of type ``MixedCut`` and their tracks will be interpreted as individual channels.
+    The output shape is ``(batch, channel, time)``.
+    The cuts will be padded with silence if necessary.
+    """
+    assert all(cut.has_recording for cut in cuts)
+    assert all(isinstance(cut, MixedCut) for cut in cuts)
+    
+    # TODO: how to ensure that each track is synced across batches?  i.e. dim=1 is the track index
+    #  and should correspond to the same mic across batches 
+
+    cuts = maybe_pad(cuts)
+    cuts = remove_pad_tracks(cuts)
```

Comment:
I think there is a pitfall here, what if a MixedCut looks like:

```
|-------cut1-------||---padding---||----cut2----|
```

or any variation of the situation where padding is in between of two cuts. I don't think Lhotse would handle these situations well with your current code. Maybe you should try only removing the padding at the end (and beginning, but for that one you have to be careful about modifying the offsets on the remaining tracks). Rather than manually removing PaddingCuts, I suggest using `.truncate()` with carefully computed offset and duration arguments; that method will handle a lot of pitfalls and edge-cases.